crates/bws/src/main.rs

```diff
@@ -404,6 +409,102 @@ async fn process_commands() -> Result<()> {
             }
         }
 
+        Commands::Run {
+            command,
+            shell,
+            no_inherit_env,
+            project_id,
+        } => {
+            let shell = match std::env::consts::OS {
+                "windows" => shell.unwrap_or_else(|| "powershell".to_string()),
+                _ => shell.unwrap_or_else(|| "sh".to_string()),
+            };
+
+            if which(&shell).is_err() {
+                eprintln!("Error: shell '{}' not found", shell);
+                std::process::exit(1);
+            }
+
+            let user_command = if command.is_empty() {
+                if atty::is(Stream::Stdin) {
+                    eprintln!("{}", Cli::command().render_help().ansi());
+                    std::process::exit(1);
+                }
+
+                let mut buffer = String::new();
+                std::io::stdin().read_to_string(&mut buffer)?;
+                buffer
+            } else {
+                command.join(" ")
+            };
+
+            if user_command.is_empty() {
+                let mut cmd = Cli::command();
+                eprintln!("{}", cmd.render_help().ansi());
+                std::process::exit(1);
+            }
+
+            let res = if let Some(project_id) = project_id {
+                client
+                    .secrets()
+                    .list_by_project(&SecretIdentifiersByProjectRequest { project_id })
+                    .await?
+            } else {
+                client
+                    .secrets()
+                    .list(&SecretIdentifiersRequest { organization_id })
+                    .await?
+            };
+
+            let secret_ids = res.data.into_iter().map(|e| e.id).collect();
+            let secrets = client
+                .secrets()
+                .get_by_ids(SecretsGetRequest { ids: secret_ids })
+                .await?
+                .data;
+
+            let environment = secrets
+                .into_iter()
+                .map(|s| (s.key, s.value))
+                .collect::<std::collections::HashMap<String, String>>();
+
+            for key in environment.keys() {
+                if !is_valid_posix_name(key) {
+                    eprintln!(
+                        "Warning: secret '{}' does not have a POSIX-compliant name",
+                        key
+                    );
+                }
+            }
+
+            let mut command = process::Command::new(shell);
+            command
+                .arg("-c")
+                .arg(&user_command)
+                .stdout(process::Stdio::inherit())
+                .stderr(process::Stdio::inherit());
+
+            if no_inherit_env {
+                let path = std::env::var("PATH").unwrap_or_else(|_| "/bin:/usr/bin".to_string());
+                command.env_clear();
+                command.env("PATH", path); // PATH is always necessary
+                command.envs(&environment);
+            } else {
+                command.envs(&environment);
```

Comment:
for additional security, do we want to remove "BWS_ACCESS_TOKEN" from inherited env variables ?

Reply:
I have also wondered this. I added the `--no-inherit-env` argument for a similar reason, but I suppose I could remove that variable by default.

Reply:
@bwdil thoughts ?

Reply:
@mzieniukbw please unpack what you mean by 'for additional security' what is your concern? About having `BWS_ACCESS_TOKEN` inherited and how it gets subsequently used thereafter? 

What are the security gains you're trying to achieve and what's the risk?

Reply:
@bwdil My though is that whatever get's executed as part of `bws run` is probably something that needs to run one off and just need read access to some of the secrets.
With `BWS_ACCESS_TOKEN` we expose much more than intended:
- Access to all secrets that the token have access to - `bws run` can specify a project, but with access token we have access to every single of them
- With access token you can do write operations - which is really bad in case it get's to wrong hands.

Reply:
@mzieniukbw if `BWS_ACCESS_TOKEN` is not required for subsequent commands that get spawned then it can be cleared from env to limit exposure. I think your concern is something that `bws run` invokes (another process) getting hijacked and then stealing the access token.

Reply:
`BWS_ACCESS_TOKEN` is now unset by default.